### ISSUE TITLE
`astro:db`: TimeStamps stored using `NOW` (in UTC) are being translated to their Local Time value making the JavaScript value of the `Date` object inaccurate

### ISSUE THREAD

**Issue by timsexperiments** [withastro/astro#10964]
### Astro Info

```block
Astro                    v4.7.1
Node                     v22.0.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             astro:db
                         @astrojs/db/file-url
                         @astrojs/tailwind
```

### Describe the Bug

Date values created with the `NOW` constant from `astro:db` (LibSQL value `CURRENT_TIMESTAMP`) are being treated as the local date value from the tables that use `Date` columns. Because the timestamp in the database is not stored with any timestamp info, the date will be treated as the local time of the client parsing the value.

### What's the expected result?

Personally I would expect one of two things:

Option 1: I would expect the date to be stored in whatever timezone it was created in (I think this might be impossible due to the way that they are stored - but I am no SQLite expert so I could be wrong)

Option 2: The date should be returned as a string (not parsed by astro, similar to the example when using the code `db.select({ date: NOW })...`), and the user should be responsible for adding the correct timestamp.

### Link to Minimal Reproducible Example

https://github.com/timsexperiments/astro-db-now-reproduction

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Yeah the fact that it is stored without the timezone is a bit of a problem. Going to discuss internally about what we should do about this, thanks for bringing it to our attention.
